deployment/constructs/dynamodb-table.ts

```diff
@@ -0,0 +1,77 @@
+import { Construct } from "constructs";
+import { RemovalPolicy } from "aws-cdk-lib";
+import * as dynamodb from "aws-cdk-lib/aws-dynamodb";
+
+interface DynamoDBTableProps {
+  readonly stage: string;
+  readonly isDev: boolean;
+  readonly name: string;
+  readonly partitionKey: { name: string; type: dynamodb.AttributeType };
+  readonly sortKey?: { name: string; type: dynamodb.AttributeType };
+  readonly lsi?: {
+    indexName: string;
+    sortKey: { name: string; type: dynamodb.AttributeType };
+  }[];
+  readonly gsi?: {
+    indexName: string;
+    partitionKey: { name: string; type: dynamodb.AttributeType };
+  };
+}
+
+export interface DynamoDBTableIdentifiers {
+  /** The invariant identifier for the table. Example: "FormAnswers" */
+  id: string;
+  /** The name of the table within the environment. Example: "production-form-answers" */
+  name: string;
+  /** The table's TableArn */
+  arn: string;
+  /** The table's TableStreamArn (if it has one) */
+  streamArn: string | undefined;
+}
+
+export class DynamoDBTable extends Construct {
+  public readonly table: dynamodb.Table;
+  public readonly identifiers: DynamoDBTableIdentifiers;
+
+  constructor(scope: Construct, id: string, props: DynamoDBTableProps) {
+    super(scope, id);
+
+    const tableName = `${props.stage}-${props.name}`;
+    this.table = new dynamodb.Table(this, "Table", {
+      tableName,
+      partitionKey: props.partitionKey,
+      sortKey: props.sortKey,
+      billingMode: dynamodb.BillingMode.PAY_PER_REQUEST,
+      stream: dynamodb.StreamViewType.NEW_AND_OLD_IMAGES,
+      pointInTimeRecoverySpecification: {
+        pointInTimeRecoveryEnabled: true,
+      },
+      removalPolicy: props.isDev ? RemovalPolicy.DESTROY : RemovalPolicy.RETAIN,
+    });
+
+    this.identifiers = {
+      id,
+      name: tableName,
+      arn: this.table.tableArn,
+      streamArn: this.table.tableStreamArn,
+    };
+
+    if (props.lsi) {
```

Comment:
Interesting that we have both a need for LSI and GSI. I wonder if that's legit or if someone didn't know the difference. Not really a point worth taking up here since the task (convert to cdk) is unrelated to that choice. Just genuinely wondering.

Reply:
I would argue that LSI and GSI both have their use cases.  However MFP is using LSI, not GSI.  I would like to get to the point that our constructs are consistent across our apps, not stripped down to what they use exactly.  That's my preference.  I still would like some sort of system to ensure consistency between our apps with code that should be the same.